### PR TITLE
Deprecate usage of `begin` as a delay and provide a `delay` attribute, instead.

### PR DESCRIPTION
**Pull request by msimpson**
Start to address the concerns raised in #1401.

### CODE REVIEW

File: src/core/a-animation.js [hunk 1]
```diff
@@ -158,6 +158,12 @@ module.exports.AAnimation = registerElement('a-animation', {
     update: {
       value: function () {
         var data = this.data;
+        // Deprecation warning for begin when used as a delay.
+        if (data.begin !== '' && !isNaN(data.begin)) {
+          console.warn("Using 'begin' to specify a delay is deprecated. Use 'delay' instead.");
+          data.delay = cloneValue(data.begin);
```
Comment (ngokevin) [inline src/core/a-animation.js]: Don't think you need to clone because it's just a number.

File: src/core/a-animation.js [hunk 2]
```diff
@@ -161,7 +161,7 @@ module.exports.AAnimation = registerElement('a-animation', {
         // Deprecation warning for begin when used as a delay.
         if (data.begin !== '' && !isNaN(data.begin)) {
           console.warn("Using 'begin' to specify a delay is deprecated. Use 'delay' instead.");
-          data.delay = cloneValue(data.begin);
+          data.delay = data.begin;
```
Comment (msimpson) [inline src/core/a-animation.js]: Gotcha.

File: src/core/a-animation.js [hunk 3]
```diff
@@ -114,7 +114,7 @@ module.exports.AAnimation = registerElement('a-animation', {
           self.count = repeat === Infinity ? 0 : parseInt(data.repeat, 10);
         }
 
-        if (isNaN(begin)) { begin = 0; }
+        if (isNaN(delay)) { delay = 0; }
```
Comment (msimpson) [inline src/core/a-animation.js]: Actually, is it safe to remove this check as well?

Comment (ngokevin) [inline src/core/a-animation.js]: Maybe good to leave in to handle weird input

Comment (msimpson) [inline src/core/a-animation.js]: Okay.

**Pull request merged**